src/include/platform/LegacyDeviceInstanceInfoProvider.h

```diff
@@ -0,0 +1,185 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <lib/core/CHIPError.h>
+#include <lib/support/Span.h>
+
+namespace chip {
+namespace DeviceLayer {
+
+constexpr uint32_t kInvalidPasscodes[] = { 00000000, 11111111, 22222222, 33333333, 44444444, 55555555,
+                                           66666666, 77777777, 88888888, 99999999, 12345678, 87654321 };
+
+class LegacyDeviceInstanceInfoProvider
```

Comment:
The idea was to do something similar to `CommissionableDataProvider`. Apart from the interface, there's also `LegacyTemporaryCommissionableDataProvider` that implements the interface by the names of `ConfigurationManager::ReadConfigValue` methods. We don't need `Legacy` in the name of the interface, as the interface will stay with us even for long :)

Reply:
Ok, I didn't know about `LegacyTemporaryCommissionableDataProvider` :)